server/embed/config.go

```diff
@@ -759,6 +765,10 @@ func (cfg *Config) Validate() error {
 		return fmt.Errorf("setting experimental-enable-lease-checkpoint-persist requires experimental-enable-lease-checkpoint")
 	}
 
+	if cfg.ExperimentalCompactHashCheckTime <= 0 {
```

Comment:
I think we should only apply this check when `ExperimentalCompactHashCheckEnabled` is `true`. In other words, we shouldn't check `ExperimentalCompactHashCheckTime` at all if `ExperimentalCompactHashCheckEnabled` is disabled.

Reply:
ack

This isn't a big deal, feel free to keep it as it's.